src/main/kotlin/org/rust/ide/refactoring/introduceVariable/impl.kt

```diff
@@ -127,3 +129,13 @@ private fun findAnchor(expr: PsiElement): PsiElement? {
 
     return anchor
 }
+
+private fun findAnchor(exprs: List<PsiElement>, chosenExpr: RsExpr): PsiElement? {
+    var anchor = findAnchor(exprs.minBy { it.startOffset } ?: chosenExpr)
+        ?: return null
+
+    while (exprs.any { !PsiTreeUtil.isAncestor(anchor.parent, it, true) }) {
+        anchor = anchor.parent
+    }
+    return anchor
```

Comment:
Looks like the following version does the same but faster
```kotlin
val parent = PsiTreeUtil.findCommonParent(anchor.parent, *exprs.toTypedArray())
while (anchor.parent != parent) {
    anchor = anchor.parent
}
```

Also, maybe it's worth to extract `while` loop into separate function because we have the same code in `findAnchor` above

Reply:
Yes, thank you, updated